src/js/jsx/sections/libraries/LibraryList.jsx

```diff
@@ -71,10 +72,23 @@ define(function (require, exports, module) {
                 command: null
             };
         },
+        
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this.props.libraries !== nextProps.libraries ||
+                   this.props.selected !== nextProps.selected ||
+                   this.props.disabled !== nextProps.disabled ||
+                   this.props.onLibraryChange !== nextProps.onLibraryChange ||
+                   !_.isEqual(this.state, nextState);
+        },
 
         componentDidUpdate: function () {
-            if (this.refs.input) {
-                this.refs.input.getDOMNode().focus();
+            var renameInput = this.refs.input;
+            
+            if (renameInput) {
+                window.setTimeout(function () {
+                    renameInput.acquireFocus();
+                    renameInput.getDOMNode().focus();
+                }, 100);
```

Comment:
This is a bummer, but I understand that there are weird focus issues going on here. We might want to increase this timeout just a little more though to be on the safe side?